File: ICT2022Y - Operating Systems 2024/C projects/factorial.c
#include <stdio.h>
#include <stdlib.h>


int factorial(int n){
    int result=1;
    while(n>0){
        result*=n;
        n--;
    }
    return result;
}
  
  
int fact(int n){
    if(n<=1){
        return 1;
    }else{
        return n*fact(n-1);
    }
}

int main(){
    int num;
    printf("Enter a number: ");
    scanf("%d",&num);
    printf("%d\n",factorial(num));
    printf("%d\n",fact(num));
}

File: ICT 2020 - Object-Oriented Techniques/Test prep/Factorial.java
import java.util.Scanner;
public class Factorial{
    public static long factorial(long n){
       if(n>1){
        return n*factorial(n-1);
       }else if(n==0 || n==1){
        return 1;
       }else{
        System.err.println("The factorial of "+n+" is invalid.");
        System.exit(1);
        return 1;
       }

    }


    public static void main(String[] args){

        Scanner scan = new Scanner(System.in);
        System.out.println("__FACTORIAL__");
        System.out.println("Enter a number: ");
        long num = scan.nextLong();
        System.out.println(factorial(num));
    }
}